pandas/tests/io/test_sql.py

```diff
@@ -3036,426 +3308,337 @@ def dtype_backend_expected(self, storage, dtype_backend) -> DataFrame:
-        res2 = sql.read_sql_table("test_schema_public_explicit", self.conn)
-        tm.assert_frame_equal(df, res2)
-        res3 = sql.read_sql_table(
-            "test_schema_public_explicit", self.conn, schema="public"
-        )
-        tm.assert_frame_equal(df, res3)
-        res4 = sql.read_sql_table("test_schema_other", self.conn, schema="other")
-        tm.assert_frame_equal(df, res4)
-        msg = "Table test_schema_other not found"
-        with pytest.raises(ValueError, match=msg):
-            sql.read_sql_table("test_schema_other", self.conn, schema="public")
+    # read dataframes back in
+    res1 = sql.read_sql_table("test_schema_public", conn)
+    tm.assert_frame_equal(df, res1)
+    res2 = sql.read_sql_table("test_schema_public_explicit", conn)
+    tm.assert_frame_equal(df, res2)
+    res3 = sql.read_sql_table(
+        "test_schema_public_explicit", conn, schema="public"
+    )
+    tm.assert_frame_equal(df, res3)
+    res4 = sql.read_sql_table("test_schema_other", conn, schema="other")
+    tm.assert_frame_equal(df, res4)
+    msg = "Table test_schema_other not found"
+    with pytest.raises(ValueError, match=msg):
+        sql.read_sql_table("test_schema_other", conn, schema="public")
 
-        # different if_exists options
+    # different if_exists options
 
-        # create a schema
-        with self.conn.begin():
-            self.conn.exec_driver_sql("DROP SCHEMA IF EXISTS other CASCADE;")
-            self.conn.exec_driver_sql("CREATE SCHEMA other;")
+    # create a schema
+    with conn.connect() as con:
+        with con.begin():
+            con.exec_driver_sql("DROP SCHEMA IF EXISTS other CASCADE;")
+            con.exec_driver_sql("CREATE SCHEMA other;")
 
-        # write dataframe with different if_exists options
-        assert (
-            df.to_sql(
-                name="test_schema_other", con=self.conn, schema="other", index=False
-            )
-            == 2
+    # write dataframe with different if_exists options
+    assert (
+        df.to_sql(
+            name="test_schema_other", con=conn, schema="other", index=False
         )
+        == 2
+    )
+    df.to_sql(
+        name="test_schema_other",
+        con=conn,
+        schema="other",
+        index=False,
+        if_exists="replace",
+    )
+    assert (
         df.to_sql(
             name="test_schema_other",
-            con=self.conn,
+            con=conn,
             schema="other",
             index=False,
-            if_exists="replace",
-        )
-        assert (
-            df.to_sql(
-                name="test_schema_other",
-                con=self.conn,
-                schema="other",
-                index=False,
-                if_exists="append",
-            )
-            == 2
-        )
-        res = sql.read_sql_table("test_schema_other", self.conn, schema="other")
-        tm.assert_frame_equal(concat([df, df], ignore_index=True), res)
-
-        # specifying schema in user-provided meta
```

Comment:
This until the end of the function doesn't even run on main so I just deleted to simplify